apis/v1alpha1/health_check.go

```diff
@@ -0,0 +1,98 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HealthCheckSpec defines the desired state of HealthCheck.
+//
+// A complex type that contains information about one health check that is associated
+// with the current Amazon Web Services account.
+type HealthCheckSpec struct {
+
+	// A complex type that contains settings for a new health check.
+	// +kubebuilder:validation:Required
+	HealthCheckConfig *HealthCheckConfig `json:"healthCheckConfig"`
+	// A complex type that contains a list of the tags that you want to add to the
+	// specified health check or hosted zone and/or the tags that you want to edit
+	// Value for.
+	//
+	// You can add a maximum of 10 tags to a health check or a hosted zone.
+	Tags []*Tag `json:"tags,omitempty"`
+}
+
+// HealthCheckStatus defines the observed state of HealthCheck
+type HealthCheckStatus struct {
+	// All CRs managed by ACK have a common `Status.ACKResourceMetadata` member
+	// that is used to contain resource sync state, account ownership,
+	// constructed ARN for the resource
+	// +kubebuilder:validation:Optional
+	ACKResourceMetadata *ackv1alpha1.ResourceMetadata `json:"ackResourceMetadata"`
+	// All CRS managed by ACK have a common `Status.Conditions` member that
+	// contains a collection of `ackv1alpha1.Condition` objects that describe
+	// the various terminal states of the CR and its backend AWS service API
+	// resource
+	// +kubebuilder:validation:Optional
+	Conditions []*ackv1alpha1.Condition `json:"conditions"`
+	// A unique string that you specified when you created the health check.
+	// +kubebuilder:validation:Optional
+	CallerReference *string `json:"callerReference,omitempty"`
+	// A complex type that contains information about the CloudWatch alarm that
+	// Amazon Route 53 is monitoring for this health check.
+	// +kubebuilder:validation:Optional
+	CloudWatchAlarmConfiguration *CloudWatchAlarmConfiguration `json:"cloudWatchAlarmConfiguration,omitempty"`
```

Comment:
Should we consider putting this field in the spec instead?

Reply:
I am not sure, there is actually an AlarmIdentifier in the Spec under the HealthCheckConfig. The CloudWatchAlarmConfiguration is then returned in the response body of a CreateHealthCheck request. So i think it is ok to have it in the Status.